Trap Quest.inform/Inform/Extensions/Every Turn/Periodic Effects.i7x

```diff
@@ -207,6 +207,54 @@ To compute stat healing:
 		decrease temp_int_dam by 1;
 		say "[bold type]You feel as though your mind is slightly clearer![roman type]";
 
+A time based rule (this is the audible jiggling rule):
+	if a random number between 1 and 100 < 50 + unlucky:
+		if debugmode > 0, say "AUDIBLE CHECK. ";
+		let audibly jiggling clothing be nothing;
+		repeat with C running through worn cursed blandness clothing:
+			if C is overdress or C is bikini top or C is bra:
+				If C is chestless or fully exposing, decide on 0;
+				If the largeness of breasts <= 2,decide on 0;
+				if C is overdress and support of (C - a bra) is >= 2, decide on 0;
```

Comment:
This is just a mess!

You can't use `(c - a bra)` in a command like this. The only place it's appropriate to use this type of notation is when declaring a polymorphic function, for example `To decide which number is the initial outrage of (C - a bra):`

Also you can't write `is >=`, you just write `>=`.

Furthermore if `C` is overdress, as the first part of the condition requires, then trying to check the `support` of `C` will trigger errors because `support` is an attribute of bras.

You are probably trying to make it so that bras with a lot of support aren't chosen. For that, you would do

```if C is bra and the support of C >= 2, decide on 0;```

And that's assuming that your `decide on 0;` actually works as intended.

Reply:
I was trying to kill two birds with one stone. I didn't think it would be immersive to have it trigger when there was a bra with a lot of support underneath or for it to be the bra. It's supposed to check for the bra that the player is currently wearing and make sure it doesn't have support more than or equal to 2. I'm going to change the line to what you suggested, though do you have an idea to how that can work?

Reply:
We'd need to first check for a bra. Since we know that the player can only wear one bra at a time, the shortest way to do it would be like this:

`if there is a worn bra and the support of a random worn bra >= 2`

But I *mostly* tend to avoid doing this because every now and then I have found that the interpreter appears to mess up and not successfully obtain the correct object and return the correct attribute (in this case, `support`). So I prefer, just to be safe, to do it like this:

```
let B be a random worn bra;
if B is bra and the support of B >= 2
```

Having said that I just checked my code and you can see in lines 1258 and 1259 of Crafting\Alchemy Products.i7x there is a (hopefully lol) working example of doing it the first way.